src/plugins/walletConnectToDapps/components/header/AddressLinks.tsx

```diff
@@ -1,63 +1,76 @@
-import { HStack, Link, Stack } from '@chakra-ui/react'
+import { AvatarGroup, Flex, HStack, Stack } from '@chakra-ui/react'
 import type { AssetId } from '@shapeshiftoss/caip'
 import { fromAccountId } from '@shapeshiftoss/caip'
 import { useMemo } from 'react'
 
+import { AssetIcon } from '@/components/AssetIcon'
 import { InlineCopyButton } from '@/components/InlineCopyButton'
 import { MiddleEllipsis } from '@/components/MiddleEllipsis/MiddleEllipsis'
 import { Text } from '@/components/Text'
-import { isSome } from '@/lib/utils'
 import { accountIdToFeeAssetId } from '@/lib/utils/accounts'
-import { selectAssetById } from '@/state/slices/selectors'
-import { useAppSelector } from '@/state/store'
 
 type AddressAndChainProps = {
   accountIds: string[]
 }
 
-const AddressLink = ({ feeAssetId, accountId }: { feeAssetId: AssetId; accountId: string }) => {
-  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId ?? ''))
-  const { account: address } = fromAccountId(accountId)
-  if (!feeAsset) return null
-  return (
+type AddressWithNetworksProps = {
+  address: string
+  chainIds: AssetId[]
+}
+
+const AddressWithNetworks = ({ address, chainIds }: AddressWithNetworksProps) => (
+  <Flex gap='2'>
+    <Flex minWidth='72px' justifyContent='flex-start'>
+      <AvatarGroup size='xs' max={3}>
+        {chainIds.map((chainId, i) => (
+          <AssetIcon key={chainId} assetId={chainId} zIndex={i} showNetworkIcon />
```

Comment:
@reallybeard leaving this for posteriority but I rage quitted after one hour on this - look like `<AvatarGroup />` in v2 rugs z-indexes. Similarly broken in prod in `Networks` section.

Reply:
commit name checks out 0e0797f53